python/ray/data/preprocessors/scaler.py

```diff
@@ -364,13 +426,26 @@ def column_robust_scaler(s: pd.Series):
 
             return (s - s_median) / diff
 
-        df.loc[:, self.columns] = df.loc[:, self.columns].transform(
-            column_robust_scaler
-        )
+        df[self.output_columns] = df[self.columns].transform(column_robust_scaler)
         return df
 
     def __repr__(self):
         return (
             f"{self.__class__.__name__}(columns={self.columns!r}, "
-            f"quantile_range={self.quantile_range!r})"
+            f"quantile_range={self.quantile_range!r}), "
+            f"output_columns={self.output_columns!r}"
         )
+
+
+def _derive_and_validate_output_columns(
```

Comment:
this method can be moved to the preprocessor class, this way all preprocessor can use it. I kept it here for now, but if we approve this approach, then I can move it in follow up PRs (when more preprocessors are extended to support append mode)